docs/usage.md

```diff
@@ -0,0 +1,325 @@
+  Elastic [Watcher](https://www.elastic.co/guide/en/kibana/current/watcher-ui.html).
+  The Elasticsearch API censors passwords from Watcher users by default, so if
+  you want to load dumped watchers into another cluster you need to provide the
+  password for the Watcher user separately.
+
+#### `log`:
+
+- `log.level`: sets the global log level for the application; accepts values
+  like "INFO" or "warning"
+- `log.ecs`: formats the log messages in [Elastic Common
+  Schema](https://www.elastic.co/guide/en/ecs/current/index.html)
+  format for easy processing in Elasticsearch.
+
+### Profiles
+
+Stacker also provides a notion of *configuration profiles*. These go in the
+`profiles` key at the top level of the config file. Each entry in the `profiles`
+section has a name, and can contain within it all the same configuration options
+as `defaults`. If you invoke `stacker` with the `--profile <name>` or `-p <name>` flags, the options you put in the named profile will override the
+defaults. For example, if your configuration looks like:
+
+```yaml
+default:
+  client:
+    timeout: 10
+
+profiles:
+  prod:
+    client:
+      timeout: 30
+```
+
+In this situation, Stacker will set the client timeout to 10 seconds, unless it
+is run with the `--profile prod` flag in which case it will set the timeout to
+30 seconds.
+
+It is perfectly acceptable to set options in a profile which you haven't set in
+the defaults; the options will be merged together when Stacker is run.
+
+> :information: Options given at the command line, like `--data-directory`
+> override both the defaults and the profile.
+
+## Command Line Usage
+
+This section describes how to use Stacker's command-line flags effectively.
+
+### Subcommands
+
+The Stacker command requires a subcommand. You can see a list of all of them by
+running `stacker` with no arguments. Most of these subcommands work the same
+way: they have a `load` and a `dump` subcommand which exports or imports that
+type of data into the configured data directory. For example:
+
+```bash
+stacker pipelines dump
+stacker saved-objects load
+```
+
+These subcommands accept a number of flags which will be discussed more in the
+next section.
+
+There are a few other subcommands which behave differently:
+
+- `system-dump`and `system-load`: iterate over all types of resources Stacker knows
+  about, and dump or load all of them.
+- `profile`: show the current configuration profile, after all the various
+  levels of configuration have been merged together.
+- `version`: Show the installed version of the `elastic-stacker` package.
+
+### Flags and Parameters
+
+In general, the individual resource subcommands (like `transforms` or
+`component-templates`) accept most of the same parameters. If you want to see
+what flags a particular subcommand takes, you can run `stacker --help <subcommand>` to see what flags it takes. (One exception to this is the `load`
```

Comment:
This is helpful, I've found it hard to intuit the help structures of the various commands the way my brain expects them. E.g. my brain wants to do `stacker <subcommand> --help` but that leads to a stack trace IIRC.

I assume this is just the way that `fire` works for handling these?

Reply:
This is partially an issue with `fire` and partially a flawed design decision on my part -- all the commands that lead to a stack trace are ones that accept `**kwargs`, and Fire interprets *anything* after the subcommand as a keyword argument, even `--help` as a keyword argument to pass into the function.

Those functions need `**kwargs` right now because not all of them accept the same flags, and the `system_dump` and `system_load` methods currently call all the individual load and dump methods with the same arguments -- this is an area of future work.